lib/OptimizationMOI/src/OptimizationMOI.jl

```diff
@@ -336,12 +372,19 @@ function SciMLBase.__solve(prob::OptimizationProblem,
         minimum = NaN
         opt_ret = :Default
     end
-    return SciMLBase.build_solution(prob,
+    return SciMLBase.build_solution(cache.prob,
```

Comment:
What would be even better is to avoid all those nested field access in the solution interfaces, like `A.prob.p` and use something like `get_p(A)`, so that anything that implements this interface can be used, e.g. a cache.